packages/main/src/plugin/index.ts

```diff
@@ -292,6 +297,27 @@ export class PluginSystem {
       return providerRegistry.installProvider(providerInternalId);
     });
 
+    ipcMain.handle(
+      'provider-registry:runPreflightChecks',
```

Comment:
we should add `Install` somewhere in the method name to be sure that it's a preflight about the installation part, not like the update part or something else